main.tf

```diff
@@ -14,6 +14,18 @@
  * limitations under the License.
  */
 
+provider "null" {
```

Comment:
Rather than actually configuring providers in reusable modules, we should just [specify the required provider versions](https://www.terraform.io/docs/configuration/terraform.html#specifying-required-provider-versions) in `versions.tf`.